core/domain/learner_playlist_services.py

```diff
@@ -0,0 +1,240 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for the learner playlist feature of the learner dashboard."""
+
+from core.domain import subscription_services
+from core.domain import user_domain
+from core.platform import models
+
+import feconf
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT = (
+    feconf.MAX_LEARNER_PLAYLIST_ACTIVITY_COUNT)
+
+
+def get_learner_playlist_from_model(learner_playlist_model):
+    """Returns the learner playlist domain object given the learner playlist
+    model loaded from the datastore.
+
+    Args:
+        learner_playlist_model: LearnerPlaylistModel. The
+            learner playlist model from the datastore.
+
+    Returns:
+        LearnerPlaylist. The learner playlist domain object corresponding to the
+            given model.
+    """
+    return user_domain.LearnerPlaylist(
+        learner_playlist_model.id,
+        learner_playlist_model.exploration_ids,
+        learner_playlist_model.collection_ids)
+
+
+def save_learner_playlist(learner_playlist):
+    """Save a learner playlist domain object as an LearnerPlaylistModel entity
+    in the datastore.
+
+    Args:
+        learner_playlist: LearnerPlaylist. The learner playlist domain object to
+            be saved in the datastore.
+    """
+    learner_playlist_model = user_models.LearnerPlaylistModel(
+        id=learner_playlist.id,
+        exploration_ids=learner_playlist.exploration_ids,
+        collection_ids=learner_playlist.collection_ids)
+
+    learner_playlist_model.put()
+
+
+def mark_exploration_to_be_played_later(
+        user_id, exploration_id, position_to_be_inserted=None):
+    """Adds the exploration id to the learner playlist of the user at the given
+    position. If the position is not specified, the exploration gets added at
+    the end. If the exploration is one of the subscribed explorations of the
```

Comment:
I have added the explanation in the comments. Hopefully is now clarifies the purpose.